pandas/core/arrays/timedeltas.py

```diff
@@ -324,12 +325,106 @@ def _evaluate_with_timedelta_like(self, other, op):
 
     __mul__ = _wrap_tdi_op(operator.mul)
     __rmul__ = __mul__
-    __truediv__ = _wrap_tdi_op(operator.truediv)
     __floordiv__ = _wrap_tdi_op(operator.floordiv)
     __rfloordiv__ = _wrap_tdi_op(ops.rfloordiv)
 
+    def __truediv__(self, other):
+        # TODO: Decimals?
+        other = lib.item_from_zerodim(other)
+
+        if isinstance(other, (ABCSeries, ABCDataFrame, ABCIndexClass)):
+            return NotImplemented
+
+        if isinstance(other, (timedelta, np.timedelta64, Tick)):
+            other = Timedelta(other)
+            if other is NaT:
+                # specifically timedelta64-NaT
+                result = np.empty(self.shape, dtype=np.float64)
+                result.fill(np.nan)
+                return result
+
+            # otherwise, dispatch to Timedelta implementation
+            return self._data / other
+
+        elif lib.is_scalar(other):
+            # assume it is numeric
+            result = self._data / other
+            freq = None
+            if self.freq is not None:
+                # Tick division is not implemented, so operate on Timedelta
+                freq = self.freq.delta / other
+            return type(self)(result, freq=freq)
+
+        if not hasattr(other, "dtype"):
+            # e.g. list, tuple
+            other = np.array(other)
+
+        if len(other) != len(self):
+            raise ValueError("Cannot divide vectors with unequal lengths")
+
+        elif is_timedelta64_dtype(other):
+            # let numpy handle it
+            return self._data / other
```

Comment:
No.  `self._data.__div__(other)` would return `NotImplemented` if other were a `TimedeltaArray`.  This PR includes a test that covers TDA/TDA.

Reply:
Yes, but that is another level of redirection, while we know this will happen and can directly do the correct thing here? 
(I suppose `is_timedelta64_dtype` only passes through those two cases of ndarray or TimedeltaArray?)

Reply:
> I suppose is_timedelta64_dtype only passes through those two cases of ndarray or TimedeltaArray?

Yes, since we exclude Series and Index at the start.

> Yes, but that is another level of redirection, while we know this will happen and can directly do the correct thing here?

I guess we could replace `other` with `getattr(other, "_data", other)` (or if/when TimedeltaArray gets an `__array__` method, just `np.array(other)`, which would be prettier)

Reply:
In the (hopefully not too distant future), TimedeltaArray will no longer be an Index. In this case we would want to explicitly grab the `._data` out of it and proceed?

Reply:
And what's the return type here? Does this need to be wrapped in a a `type(self)` so that we return a TimedeltaArray?